packages/pwa/app/commerce-api/index.js

```diff
@@ -117,44 +110,39 @@ class CommerceAPI {
                     get: function(obj, prop) {
                         if (typeof obj[prop] === 'function') {
                             return (...args) => {
+                                const fetchOptions = args[0]
                                 const {locale, currency} = self._config
 
-                                if (args[0].ignoreHooks) {
+                                if (fetchOptions.ignoreHooks) {
                                     return obj[prop](...args)
                                 }
 
-                                return self.willSendRequest(prop, ...args).then((newArgs) => {
-                                    let {sendLocale = true, sendCurrency = false} = apiConfigs[key]
-
-                                    // By default we send the local param and don't send the currency param.
-                                    // You can modify when these are send by using a boolean value send the
-                                    // currency/locale for all calls for a given API, or define an array listing
-                                    // the API's methods you want the currency/locale information to be sent.
-                                    sendLocale = Array.isArray(sendLocale)
-                                        ? sendLocale.includes(prop)
-                                        : !!sendLocale
-                                    sendCurrency = Array.isArray(sendCurrency)
-                                        ? sendCurrency.includes(prop)
-                                        : !!sendCurrency
-
-                                    // Ensure we don't send the pseudo locale.
-                                    if (locale !== 'en-XB') {
-                                        sendLocale = false
-                                    }
-
-                                    // Inject the locale and currency to the API call via it's parameters.
-                                    //
-                                    // NOTE: The commerce sdk isomorphic will complain if you pass parameters to
-                                    // it that it doesn't expect, this is why we only add the locale and currency
-                                    // to some of the API calls.
-                                    // We use the default locale for the API calls when running the app using the
-                                    // pseudo locale 'en-XB'.
-                                    newArgs[0].parameters = {
-                                        ...newArgs[0].parameters,
-                                        ...(sendLocale ? {locale} : {}),
-                                        ...(sendCurrency ? {currency} : {})
```

Comment:
It was possible for `parameters` to already contain a locale. If `sendLocale` is false, the old implementation actually does not delete the locale property.